src/phoenix/server/api/mutations/user_mutations.py

```diff
@@ -1,34 +1,65 @@
-import asyncio
 import secrets
-from functools import partial
-from typing import Optional
+from contextlib import AsyncExitStack
+from datetime import datetime, timezone
+from typing import Literal, Optional, Tuple
 
 import strawberry
-from sqlalchemy import insert, select
+from sqlalchemy import Select, select
+from sqlalchemy.orm import joinedload
 from sqlean.dbapi2 import IntegrityError  # type: ignore[import-untyped]
+from strawberry import UNSET
+from strawberry.relay import GlobalID
 from strawberry.types import Info
 
 from phoenix.auth import (
     DEFAULT_SECRET_LENGTH,
-    compute_password_hash,
+    PASSWORD_REQUIREMENTS,
     validate_email_format,
     validate_password_format,
 )
-from phoenix.db import models
+from phoenix.db import enums, models
 from phoenix.server.api.auth import HasSecret, IsAdmin, IsAuthenticated, IsNotReadOnly
 from phoenix.server.api.context import Context
 from phoenix.server.api.input_types.UserRoleInput import UserRoleInput
+from phoenix.server.api.types.node import from_global_id_with_expected_type
 from phoenix.server.api.types.User import User
+from phoenix.server.bearer_auth import PhoenixUser
 
 
 @strawberry.input
 class CreateUserInput:
     email: str
-    username: Optional[str]
+    username: Optional[str] = UNSET
     password: str
     role: UserRoleInput
 
 
+@strawberry.input
+class PatchSelfUserInput:
+    username: Optional[str] = UNSET
+    password: Optional[str] = UNSET
+
+    def __post_init__(self) -> None:
+        if not self.username and not self.password:
+            raise ValueError("At least one field must be set")
+        if self.password:
+            PASSWORD_REQUIREMENTS.validate(self.password)
+
+
+@strawberry.input
+class PatchUserInput:
+    user_id: GlobalID
+    role: Optional[UserRoleInput] = UNSET
+    username: Optional[str] = UNSET
+    password: Optional[str] = UNSET
```

Comment:
If the password is being set I feel like we should force them to enter their old password. Otherwise anybody that gains access to the person's laptop can change their password

Reply:
What about when an admin is changing the password for the user? That's what this code is doing right now: it sets the `resset_password` flag to `True`.

Reply:
ok, i think you're referring to PathSelfUser, in which case yea, probably should require the current password

Reply:
ok. makes sense. i'll add the requirement for requester's password